File: ProyectoMysql/server/BusinessLayer/EntLista.py
from DataLayer.EntListaDB import *
from Utilidades.Conexion.configMysql import StartTransaction,EndTransaction


class EntLista:
    def Registrar(Ent: EntListaItemModel):
        try:
            StartTransaction()
            data = EntListaDB.RegistrarDB(Ent)
            EndTransaction()
            return data
        except Exception as e:
            Restore()
            print(e)

    def BuscarItem(Codigo : str, Nombre :str):
        try:
            data =  EntListaDB.BuscarItem(Codigo, Nombre)
            return data
        except Exception as e:
            print(e)     

    def ObtenerItem(ListaId : int):
        try:
            data =  EntListaDB.ObtenerItem(ListaId)
            return data
        except Exception as e:
            print(e)    


    def ObtenerItems(Codigo : str):
        try:
            data =  EntListaDB.ObtenerItems(Codigo)
            return data
        except Exception as e:
            print(e)    

File: ProyectoMysql/Api/LogisticStorage/LogisticStorage.BusinessLayer/EntLista.cs
using LogisticStorage.DataLayer;
using LogisticStorage.EntityLayer;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LogisticStorage.BusinessLayer
{
    public class EntLista
    {
  
        public static Int32 Registrar(EntListaEntity Ent)
        {
            EntListaDB DB = new EntListaDB();
            DB.Registrar(Ent);
            return Ent.ListaId;
        }
    
        public static List<EntListaEntity> BuscarItem(String Codigo, String Nombre)
        {
            EntListaDB DB = new EntListaDB();
            return DB.BuscarItem(Codigo, Nombre);
        }
    
        public static List<EntListaEntity> ObtenerItem(Int32 ListaId)
        {
            EntListaDB DB = new EntListaDB();
            return DB.ObtenerItem(ListaId);
        }

        public static List<EntListaEntity> ObtenerItems(String Codigo)
        {
            EntListaDB DB = new EntListaDB();
            return DB.ObtenerItems(Codigo);
        }
    }
}
